baseapp/baseapp.go

```diff
@@ -149,7 +147,6 @@ func NewBaseApp(
 		db:               db,
 		cms:              store.NewCommitMultiStore(db),
 		storeLoader:      DefaultStoreLoader,
-		router:           NewRouter(),
 		queryRouter:      NewQueryRouter(),
```

Comment:
Looking at this, the legacy query router can also be removed, since https://github.com/cosmos/cosmos-sdk/pull/12725. @facundomedica would you like to take care of this too?